src/api/v2/v2AzureResourcesApi.ts

```diff
@@ -148,7 +148,7 @@ export interface V2AzureResourcesApi extends AzureResourcesApiBase {
     /**
      * Show a quick picker of app resources. Set `options.type` to filter the picks.
      */
-    pickResource<TModel>(options?: ResourcePickOptions): vscode.ProviderResult<TModel>
+    pickResource<TModel extends ResourceModelBase>(actionContext: IActionContext, options?: ResourcePickOptions): vscode.ProviderResult<TModel | ApplicationResource>;
```

Comment:
If at all possible, I would really, really, *really* prefer to keep `IActionContext` out of the API.  For the most part, that interface represents implementation details of our extensions (e.g. a conveyance for telemetry properties) that I don't think should be forced upon all extensions.